pkgs/applications/networking/cluster/hadoop/default.nix

```diff
@@ -51,20 +73,30 @@ let
           computers, each of which may be prone to failures.
         '';
         maintainers = with maintainers; [ volth illustris ];
-        platforms = [ "x86_64-linux" ];
+        platforms = builtins.attrNames sha256;
       };
 
     };
 in
 {
   # Different version of hadoop support different java runtime versions
   # https://cwiki.apache.org/confluence/display/HADOOP/Hadoop+Java+Versions
-  hadoop_3_3 = common rec {
-    pname = "hadoop";
-    version = "3.3.1";
-    sha256 = "1b3v16ihysqaxw8za1r5jlnphy8dwhivdx2d0z64309w57ihlxxd";
-    untarDir = "${pname}-${version}";
-    jdk = jdk11_headless;
+  hadoop_3_3 = common
+    rec {
+      pname = "hadoop";
+      version = "3.3.1";
+      sha256 = {
+        x86_64-linux = "1b3v16ihysqaxw8za1r5jlnphy8dwhivdx2d0z64309w57ihlxxd";
+        x86_64-darwin = "1b3v16ihysqaxw8za1r5jlnphy8dwhivdx2d0z64309w57ihlxxd";
+        aarch64-linux = "sha256-v1Om2pk0wsgKBghRD2wgTSHJoKd3jkm1wPKAeDcKlgI=";
+        aarch64-darwin = "sha256-v1Om2pk0wsgKBghRD2wgTSHJoKd3jkm1wPKAeDcKlgI=";
```

Comment:
there should be some instruction on how to generate the shas

E.g. `nix-prefetch-url ...`

Reply:
Whoops! It should be fixed in <https://github.com/NixOS/nixpkgs/pull/158613/commits/0aaa8f7f530f84015f5cccbe71a6549eb8908185>.